x-pack/plugin/esql/compute/ann/src/main/java/org/elasticsearch/compute/ann/MvEvaluator.java

```diff
@@ -16,17 +16,30 @@
  * Implement an evaluator for a function reducing multivalued fields into a
  * single valued field from a static {@code process} method.
  * <p>
- *     Annotated methods can have two "shapes": pairwise processing and
- *     accumulator processing. Pairwise is <strong>generally</strong>
- *     simpler and looks like {@code int process(int current, int next)}.
- *     Use it when the result is a primitive. Accumulator processing is
- *     a bit more complex and looks like {@code void process(State state, int v)}
- *     and it useful when you need to accumulate more data than fits
- *     in a primitive result. Think Kahan summation.
+ *     Annotated methods can have three "shapes":
  * </p>
+ * <ul>
+ *     <li>pairwise processing</li>
+ *     <li>accumulator processing</li>
+ *     <li>position at a time processing</li>
+ * </ul>
  * <p>
- *     Both method shapes support at {@code finish = "finish_method"} parameter
- *     on the annotation which is used to, well, "finish" processing after
+ *     Pairwise processing is <strong>generally</strong> simpler and looks
+ *     like {@code int process(int current, int next)}. Use it when the result
+ *     is a primitive.</p>
+ * <p>
+ *     Accumulator processing is a bit more complex and looks like
+ *     {@code void process(State state, int v)} and it useful when you need to
+ *     accumulate more data than fits in a primitive result. Think Kahan summation.
+ * </p>
+ * <p>
+ *     Position at a time processing just hands the block, start index, and end index
```

Comment:
Nice. I think this should make a MV function returning the element at a position easy now.